orange_tree2.rb

```diff
@@ -0,0 +1,108 @@
+class OrangeTree
+  attr_reader :height, :age, :orange_count, :death_age
+
+  # creates new tree
+  def initialize
+    @height = 1
+    @age = 0
+    @orange_count = 0
+    #random age between 90-110
+    @death_age = (rand(90..110))
+    @max_height = 30
+  end
+
+  def increases_height
+    #increases height according to age.
+    if (@age <= 5)
+      @height += 3
+      #puts "#{@height}"
+    elsif (@age <= 15)
+      @height += 1
+    elsif (@age <= 40) && (@height < @max_height)
+      @height += 0.5
+    end
+  end
+
+  def produces_oranges
+    #produces orange? according to age
+    if (@age < 6)
+      @orange_count = 0
+    elsif (@age >= 6) && (@age < 20)
+      @orange_count = (rand(50..100))
+    elsif (@age >= 20) && (@age < 40)
+      @orange_count = (rand(100..200))
+    elsif (@age >= 40) && (@age < 60)
+      @orange_count = (rand(200..300))
+    else
+      @orange_count = (rand(100..200))
+    end
+  end
+
+  def one_year_passes
+    #increases age by one year
+    @age += 1
+    increases_height
+    produces_oranges
+    #is alive?
+    if (@age >= @death_age)
+      puts "Tree has passed on. Good-bye, tree."
+    end
+  end
+
+  def pick_an_orange
+    @orange_count -= 1
+    if (@orange_count <= 0)
+      puts "Sorry, tree is empty!"
```

Comment:
Can you go into detail about the difference between return and puts and explain why one is more necessary than the other? Without background/explanation I don't fully understand.